prediction_market_agent/agents/microchain_agent/blockchain/contract_class_converter.py

```diff
@@ -0,0 +1,143 @@
+import json
+from typing import Any
+
+import requests_cache
+from eth_typing import ChecksumAddress
+from microchain import Function
+from prediction_market_agent_tooling.tools.contract import ContractOnGnosisChain
+from web3 import Web3
+
+from prediction_market_agent.agents.microchain_agent.blockchain.code_interpreter import (
+    CodeInterpreter,
+)
+from prediction_market_agent.agents.microchain_agent.blockchain.models import (
+    AbiItemTypeEnum,
+    ABIMetadata,
+)
+
+TYPE_MAPPING: dict[str, (str, Any)] = {
+    "address": (
+        "str",
+        Web3.to_checksum_address("0xe91d153e0b41518a2ce8dd3d7944fa863463a97d"),
+    ),  # Solidity address to Python string
+    "uint": ("int", 1),  # Solidity unsigned integer to Python int
+    "uint8": ("int", 1),  # Solidity unsigned integer to Python int
+    "uint256": ("int", 1),  # Solidity unsigned integer (256-bit) to Python int
+    "int": ("int", 1),  # Solidity integer to Python int
+    "bool": ("bool", True),  # Solidity boolean to Python bool
+    "string": ("str", "test string"),
+    # Add more mappings as needed
+}
+
+
+class ClassFactory:
+    def create_class(self, class_name, base_classes=(), attributes=None):
+        if attributes is None:
+            attributes = {}
+
+        new_class = type(class_name, base_classes, attributes)
+        return new_class
+
+
+class ContractClassConverter:
```

Comment:
I could be wrong, but it looks like we can get a lot of the abi stuff for free from `Web3.eth.contract`

- https://web3py.readthedocs.io/en/stable/web3.contract.html#web3.contract.Contract
- https://web3py.readthedocs.io/en/stable/web3.contract.html#web3.contract.ContractFunction
- https://chatgpt.com/share/85ccdb38-bfe0-424f-ba2a-f6240a79c941

though the web3 `ContractFunction` -> Microchain Function we still have to do ourselves

Reply:
Per our discussion during stand-up, web3py offers limited functionality with regards to converting ABI into types and/or function definitions, hence can't be leveraged for this.
Note that we are using web3py under the hood inside PMAT (dynamic function calls PMAT -> PMAT calls web3py.contract).